src/base/net/downloadmanager.cpp

```diff
@@ -304,7 +309,11 @@ void Net::DownloadManager::processRequest(DownloadHandlerImpl *downloadHandler)
     QNetworkReply *reply = m_networkManager->get(request);
     connect(reply, &QNetworkReply::finished, this, [this, downloadHandler]
     {
-        handleDownloadFinished(downloadHandler);
+        const ServiceID id = ServiceID::fromURL(downloadHandler->url());
```

Comment:
You need to check whether this URL is actually registered as a sequential service. At least, you shouldn't use `operator[]` below, because it will unexpectedly add this URL in sequential services registry if it wasn't there.

Reply:
Good notice, Fixed with 7f6556f1703d81227435d7ef755c2bba39315d60
I added both a check for containing the key to avoid unneeded timers.
Also added default 0s value in any case.